pkg/queuesystem/queue_system.go

```diff
@@ -0,0 +1,19 @@
+package queuesystem
+
+// Messages is the base struct for getting
+// data from a specified queue system
+type Messages interface {
+	GetQueueMessages() ([]QueueMessage, error)
+	DeleteMessageFromQueue(string) error
+}
+
+// QueueMessage is the abstraction for formatting
+// and using data that comes from the queues
+type QueueMessage struct {
```

Comment:
type StandardBuildRequest struct {
	RepoFolder     string `json:"repofolder"`
	ConfigPath     string `json:"configpath"`
	EngineName     string `json:"enginename"`
	EnginePlatform string `json:"engineplatform"`
	EngineVersion  string `json:"engineversion"`
	Success        bool   `json:"success"`
	Log            string `json:"log"`
	BuildrId       string `json:"buildrid"`
	BuildId        string `json:"buildid"`
	Type           string `json:"type"`
}

Could probably do with ProjectName in there too